### ISSUE TITLE
[QA] concerning the 'i' on blue circle icon

### ISSUE THREAD

**Issue by jnweiger** [owncloud/client#8970]
Seen with owncloud-client 2.9.0-rc1

This icon, 
![image](https://user-images.githubusercontent.com/1108546/131497268-6e0a3f47-b5aa-427f-95ee-6e3e78b7cbd2.png)
was introduced in 2.9.0 -- it is meant to provide additional information about the current sync state.

In the following scenarios, it achives the opposite. It creates confusion.

* Using a large account, like my personal damken-cloud account, I always see the (i), where I  formerly saw a green checkmark, after all syncing is done.
  For this account, it is normal to have several "not synced" issues.
* Not all "not synced" issues cause the (i) icon. I have a very similar situation in my cloud.owncloud.com account. There are also many "not synced" issues currently listed. But this account shows the green checkmark.
* What exactly is the meaning of the (i) icon. Is it "ignored" or "information"? Please specify.
  Traditionally, I'd associate a single lower case i with "information". But it is hard to gain any information, why it is appearing in one account, and not in the other.
* In a multi-account situation, the (i) icon remains visible in the system tray during an ongoing sync.
   - Previously the tray icon changed to circular arrows on blue during sync. This is a nice visual confirmation, that newly changed files made it into the cloud.
   - Now, the (i) in the system tray changes to a green checkmark, while the account that causes the (i) is busy syncing.
   - Now, the (i) remains in the system tray unchanged, while another account (that had a green checkmark in its on the settings page) is syncing.
   - thus it is needed to first open the owncloud settings page, click on the correct account tab. Then the user can see the arrows, while saving.

**Comment by jnweiger** [comment]
@michaelstingl Large parts can be covered by documentation.
   - E.g. create a list of "not synced" cases, and specify "good/bad/ugly" - causing nothing / blue (i) / (x).
 
But I believe, some (multi-account situation) should be improved.
  - If this behaviour works as designed, I'd first challenge the design, before sending it to documentation.

**Comment by jnweiger** [comment]
![image](https://user-images.githubusercontent.com/1108546/131521499-8f40c6ea-6b6b-4eeb-842a-4b96c7e8a7aa.png)
![image](https://user-images.githubusercontent.com/1108546/131522235-7de8f6ab-856a-4e6f-af5f-82db9952a9d1.png)
![image](https://user-images.githubusercontent.com/1108546/131522378-5320bd39-b027-4888-a887-df6f1e7ea574.png)
![image](https://user-images.githubusercontent.com/1108546/131522555-4bc82824-72f4-4ac0-824d-78adc774aaa8.png)
![image](https://user-images.githubusercontent.com/1108546/131523120-b365e839-c53f-4c0e-a6fd-5e6bb46a0e36.png)
![image](https://user-images.githubusercontent.com/1108546/131523032-fab75cc3-6724-4910-b8f5-4018d03fb04d.png)

**Comment by TheOneRing** [comment]
The I means info and I don't get your point?

**Comment by aNDy-Squirrel** [comment]
I also find this blue "i" counter-intuitive. In the owncloud window, under the activity button, the "not synced" tab has the same icon, so I see this "i" as an indication that "there is something to tell about why not all files are synced (that should be synced)". However, in the list I see only files that are not synced for a good reason, such as it is in my ignore list, it's a symbolic link, or because it's hidden. Not syncing these is normal behavior and I don't need information about that, and I would expect the green check mark in such a situation. (I guess) this also hides that situation when there is really syncing problem that needs my attention.

**Comment by TheOneRing** [comment]
We might need some fine tuning on when we report  a problem. Ignored system files are at the moment not reported as an issue.
Not syncing symlink, depends on the situation and your lvl of linux mastery.
- Your important presentation for tomorrow is just symlinked and you are about to get fired tomorrow because the client didn't sync but displayed an all Ok 😱 

Why do you have a symlink in your sync folder if you know it won't sync in the first place.

Hidden files should not cause an error report.

Maybe the original meaning of the state "Problem" should really display a yellow exclamation mark but the blue i is what the current design provides.

**Comment by aNDy-Squirrel** [comment]
I replaced the symlinks by the actual files, and have the green check mark. Great.

> Why do you have a symlink in your sync folder if you know it won't sync in the first place.
In one case it was a file that I have elsewhere on my disk, and to prevent duplicates I symlinked it. And there was no need to share it with the others. So having  the symlink was not entirely irrational, but - I agree - not a good practice.

> Your important presentation for tomorrow is just symlinked and you are about to get fired tomorrow because the client didn't sync but displayed an all Ok scream
That's why I always have a second copy of my presentations on a USB stick :-P

**Comment by avrhack** [comment]
Had to revert to 2.8.2 as 2.9.0 is unusable with this enhancement. I only have hidden and ignored files in my error list yet the 'i' icon is permanently there. Massive step backwards in usability as others have said here.

**Comment by TheOneRing** [comment]
In which way ignored?

**Comment by avrhack** [comment]
> In which way ignored?

Two ways - I have 'unticked' the 'Sync hidden files' box in the main Settings page and I have added some additional file match patterns in the dialog box under 'Edit Ignored Files'. 

That all works perfectly - has always generated entries in the 'Not Synced' tab list but on 2.8.2 and previously that tab had two tick boxes 'Show Warnings' and 'Show Ignored Files'. I simply left 'Show Ignored Files' unticked and so would see only real issues to be dealt with. And those ignored files never caused the icon to indicate anything other than success.

I notice those tick boxes are also now gone in 2.9.0 - another massive step backwards IMHO.

Basically this worked perfectly and intuitively before and now it's confusing and never shows a 'correct sync' green tick box.

**Comment by TheOneRing** [comment]
I can only repeat myself, ignored files and hidden files won't cause the i.
If you have files that failed multiple times to sync you will have the i. When the error occurs the first time you will have a red x. If the error occurs repeatedly you will get the i.

If you have symlinks that can't be synced you will get the i.

There are some more cases but:
- Hidden files won't cause the problem state.
- Ignored files won't cause the problem state.
![image](https://user-images.githubusercontent.com/200626/132717187-4cb52a0b-9ee1-4d84-b414-ffffe9ef8ea8.png)

**Comment by avrhack** [comment]
> I can only repeat myself, ignored files and hidden files won't cause the i.

And I can only repeat myself and say that they ARE! There are NO red 'x' anywhere - only ignored and hidden files, yet the 'i' icon appears.

If that's not the intended design then this is a bug in its implementation in 2.9.0.

**Comment by michaelstingl** [comment]
> If that's not the intended design then this is a bug in its implementation in 2.9.0.

Would be awesome if you could provide reliable steps to reproduce, screenshots, maybe log files, to improve here.

**Comment by avrhack** [comment]
> > If that's not the intended design then this is a bug in its implementation in 2.9.0.
> 
> Would be awesome if you could provide reliable steps to reproduce, screenshots, maybe log files, to improve here.

That's the least I can do to help, given I'm the one complaining about it LOL! I will reinstall 2.9.0 temporarily, get those data and post here. Should be done in the next 20 minutes or so......

**Comment by avrhack** [comment]
> > If that's not the intended design then this is a bug in its implementation in 2.9.0.
> 
> Would be awesome if you could provide reliable steps to reproduce, screenshots, maybe log files, to improve here.

Firstly with 2.8.2: you can see there are 1216 files not synced:

![Screenshot 2021-09-09 at 16 53 41](https://user-images.githubusercontent.com/9513181/132721109-e6637e3a-ecbd-46c7-b683-cc149c0cbf4c.png)

Unticking the 'show ignored files' results in NO errors remaining in the list as in the screenshot below:

![Screenshot 2021-09-09 at 16 54 18](https://user-images.githubusercontent.com/9513181/132721230-2dae00f5-a1cd-4e38-a03f-a07c2dc426cd.png)

However with the exact same config after installing 2.9.0, firstly there is no longer an option to drop ignored files from the error list which I think is a HUGE mistake and step backwards, but the list is still only ignored files exactly as below. However this results in the 'i' icon appearing which Hannah claimed would not happen, but it clearly is!

![Screenshot 2021-09-09 at 16 56 07](https://user-images.githubusercontent.com/9513181/132721447-b43f69ae-53fc-4ffb-8db5-dcc0c9d3a04b.png)

Unfortunately I can't upload any other logs as all the documentation links that google comes up with are broken and having searched in vain from the owncloud.com pages I can find no documentation on the client that's linked from there any more. Someone seems to have reorganised things and either broken and/or removed it.

**Comment by The9Cat** [comment]
I agree with the other posters that the new policy, while strictly correct and not a 'formal' bug, is not that helpful.  For most users, the new policy simply replaced 'green check' with 'i'.  @TheOneRing: please consider the following cases:

- Most heavy users like me will have patterns on the ignore list. 
- Similarly, I have several directories in my ownCloud root that I use for active development.   I realize that symlinks will not be backed up and I am taking advantage of that.

Both of these imply that 'i' is the new 'green check'.  My icon will always be 'i'.  So 'green check' is now useless and the client is giving me a list of 'info' conditions that I will never bother to read.

**Comment by TheOneRing** [comment]
I created https://github.com/owncloud/client/issues/9000 so you will be able to filter by the issue type.

Additionally we could talk to add an option to not display the problem state (i) on not synced sym links.

**Comment by TheOneRing** [comment]
> I agree with the other posters that the new policy, while strictly correct and not a 'formal' bug, is not that helpful. For most users, the new policy simply replaced 'green check' with 'i'. @TheOneRing: please consider the following cases:
> 
> * Most heavy users like me will have patterns on the ignore list.

This case is not supposed to cause the i state. If you have a look on my Screenshot I have ignored files and hidden files and the owncloud icon in the mac menu bar has a check mark.

> * Similarly, I have several directories in my ownCloud root that I use for active development.   I realize that symlinks will not be backed up and I am taking advantage of that.

Then you have to accept that you will get the problem state icon as we are not able to fully sync your data.
> 
> Both of these imply that 'i' is the new 'green check'. My icon will always be 'i'. So 'green check' is now useless and the client is giving me a list of 'info' conditions that I will never bother to read.

If everything syncs properly you don't have conflicts you should not see the i this is the expected scenario for a normal user.

**Comment by The9Cat** [comment]
Thanks, I get it.  And I confirm that it works exactly as you say: the 'i' is a result of symlinks not ignored and hidden files.

Stilll, I prefer the behavior of the earlier client versions.   And I endorse your suggestion of having a toggle for not flagging symlinks as a problem state.  After all, the symlink is is NOT a file but a file system feature.

**Comment by TheOneRing** [comment]
Symlinks can lead to unsynced files -> incomplete syncs.

My recommendation is to not put build folders into a sync root, if a developed application requires certain files to be present in the build folder however it a sync folder pair could be used to sync the data location.

![image](https://user-images.githubusercontent.com/200626/132853597-0cce7d64-fecf-4a9d-9583-7778f3b60de0.png)

**Comment by The9Cat** [comment]
Okay.  Got it.

But now back to the original issue: I restarted the client and I'm getting 'i' and the only reported 'states' are 'ignored' or 'hidden',  just as @avrhack reports.

So there seems to be a bug somewhere.

**Comment by avrhack** [comment]
> > I agree with the other posters that the new policy, while strictly correct and not a 'formal' bug, is not that helpful. For most users, the new policy simply replaced 'green check' with 'i'. @TheOneRing: please consider the following cases:
> > 
> > * Most heavy users like me will have patterns on the ignore list.
> 
> This case is not supposed to cause the i state. If you have a look on my Screenshot I have ignored files and hidden files and the owncloud icon in the mac menu bar has a check mark.

But as I've repeatedly said and you seem to be completely ignoring it despite the screen shots, it **DOES** cause the 'i' state in at least some cases ie mine!

**Comment by MichaKersloot** [comment]
On Windows it seems quite normal to include My Documents there are (I think) symbolic links that even the user doesn't see, but now it shows the blue (i) and for the user there is no solution to prevent this at the moment.

**Comment by avrhack** [comment]
> On Windows it seems quite normal to include My Documents there are (I think) symbolic links that even the user doesn't see, but now it shows the blue (i) and for the user there is no solution to prevent this at the moment.

Agreed. I've had to go back to 2.8.2 as 2.9.0 is unusable with this issue IMHO - despite the claim, ignored/hidden files with **NO** other errors **DOES** cause the 'i' to be shown.....

**Comment by The9Cat** [comment]
Yes, I'm just living with the 'i' state.  I suppose I could downgrade as well.

**Comment by TheOneRing** [comment]
Downgrading is not a supported scenario .

**Comment by jnweiger** [comment]
When I create one symlink, I get the i on blue circle.
Now log out and re-login the client, the i disappears and I have my green checkmark again.

On another account, where I have many symlinks (which are an esential structure of my tree), a force sync also changes the blue i into a green checkmark (not sure f it comes back after a while??)

**Comment by avrhack** [comment]
> Yes, I'm just living with the 'i' state. I suppose I could downgrade as well.

As Hannah has said it's not supported, but for me it's working perfectly so that would be my recommendation for sure. I am toying with the idea of forking the source and reverting this change but that's a very undesirable solution.

I'm still hopeful the developers will see why this needs to be reverted - both the 'i' for hidden and ignored files and also the decision to remove the extremely useful checkboxes that allowed you to filter out all hidden and ignored file messages from the issues list precisely so we could focus on **real** errors.

I really don't understand why the decision has been taken to clutter up the issues list with things you have purposely asked the client to do, and worse prevent you now from suppressing those messages.

With 2.9.0 there is a hugely increased risk that you will miss a real error as a result. Big step backwards.

**Comment by TheOneRing** [comment]
As the discussion in this ticket lost all seriousness I will now close the issue with a final wrap up.

- We are working on an enhanced issue widget which will allow filtering by the type of issue
 - You will be able to hide all hidden files issue and see what is causing the i icon for you

Syncing symlinks is not supported and never was supported, your sync will be incomplete and we will display a different icon than what you are used to.

In case of a serious issue we will still display a red icon.

**Comment by jnweiger** [comment]
We all know that ownCloud does not sync symlinks. 
So there is no reason, that a symlink cause the 'i'. 

The heated discussion was an important discussion from my POV.
We had discussed that new (mis)feature internally during early QA, but came to no conclusion, except that we agree to disagree. 

I agree with "Technically correct (according to specs) is in this case not a helpful thing."
To me it is still unclear, if the 'i' is actually behaving according to specs. That point was also discussed above.
If so, the specs should be reviewed. I believe, users are more likely to take action, if they see the 'i' coming on related to a change that they made, rather than when they have it as a permanent state, which they perceive they do not control.

**Comment by SamuAlfageme** [comment]
@TheOneRing is there an update on a potential way to handle this state on the client on a future update on the client? i.e. would it be possible to simply acknowledge the `i` icon to prevent user that want to live with conflicts, etc. to continue with a happy ✅ state? 

We're getting a few requests that indicate that this causes a bit of confusion amongst users.

**Comment by TheOneRing** [comment]
For the moment there are no plans to hide sync errors.

**Comment by jnweiger** [comment]
> On Windows it seems quite normal to include My Documents there are (I think) symbolic links that even the user doesn't see, but now it shows the blue (i) and for the user there is no solution to prevent this at the moment.

The windows case is improved in client 3.0.0 (and I believe already in 2.11) - the client now happily syncs the *.lnk files into the server. (Even if it refers to e.g. `C:\Windows\System32\...`, which opens another can of worms).

Under Linux, a symlink is still considered an error and flagged with a blue (i). 
To reiterate my initial point: The symlink should IMO either be handled the same as in windows, or at least be handled like a hidden file: Silently ignored.

For both, windows `.lnk` and Linux symlink, my predered solution would be
* if it refers to a location inside the same sync root, it should be synced.
* if it points outside, it should be silently ignored.
